cocoa/src/toga_cocoa/icons.py

```diff
@@ -38,29 +38,20 @@ def __init__(self, interface, path):
         else:
             self.path = path
 
-        try:
-            # We *should* be able to do a direct NSImage.alloc.init...(), but if the
-            # image file is invalid, the init fails, returns NULL, and releases the
-            # Objective-C object. Since we've created an ObjC instance, when the
-            # object passes out of scope, Rubicon tries to free it, which segfaults.
-            # To avoid this, we retain result of the alloc() (overriding the default
-            # Rubicon behavior of alloc), then release that reference once we're
-            # done. If the image was created successfully, we temporarily have a
-            # reference count that is 1 higher than it needs to be; if it fails, we
-            # don't end up with a stray release.
-            image = NSImage.alloc().retain()
-            self.native = image.initWithContentsOfFile(str(path))
-            if self.native is None:
-                raise ValueError(f"Unable to load icon from {path}")
-        finally:
-            # Calling `release` here disabled Rubicon's "release on delete" automation.
-            # We therefore add an explicit `release` call in __del__ if the NSImage was
-            # initialized successfully.
-            image.release()
-
-    def __del__(self):
-        if self.native:
-            self.native.release()
+        # We *should* be able to do a direct NSImage.alloc.init...(), but if the
+        # image file is invalid, the init fails, returns NULL, and releases the
+        # Objective-C object. Since we've created an ObjC instance, when the
+        # object passes out of scope, Rubicon tries to free it, which segfaults.
+        # To avoid this, we retain result of the alloc() (overriding the default
+        # Rubicon behavior of alloc), then release that reference once we're
+        # done. If the image was created successfully, we temporarily have a
+        # reference count that is 1 higher than it needs to be; if it fails, we
+        # don't end up with a stray release.
+        image = NSImage.alloc().retain()
+        self.native = image.initWithContentsOfFile(str(path))
+        if self.native is None:
+            raise ValueError(f"Unable to load icon from {path}")
+        image.release()
```

Comment:
This is an interesting one: Since a manual `release` call no longer disables Rubicon's release-on-delete logic, we can now only call `image.release()` manually when indeed a **additional** release is needed.

Reply:
I agree this is one case where manual reference management is still required. https://github.com/beeware/rubicon-objc/issues/26#issuecomment-2496141943 is a proposal for how Rubicon could handle this automatically.

Reply:
I agree this is one case where manual reference management is still required. https://github.com/beeware/rubicon-objc/issues/26#issuecomment-2496141943 is a proposal for how Rubicon could handle this automatically.